RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -151,120 +157,149 @@ namespace mkfit {
         y[ts] = tk.y();
         z[ts] = tk.z();
         d0[ts] = tk.d0BeamSpot(bspot.x, bspot.y);
+
+        b.register_entry_safe(
+            oldPhi[ts],
+            eta[ts]);  // If one is sure values are *within* axis ranges: b.register_entry(oldPhi[ts], eta[ts]);
       }
 
-      for (int ts = 0; ts < ns; ts++) {
+      b.finalize_registration();
+
+      for (int sorted_ts = 0; sorted_ts < ns; sorted_ts++) {
+        int ts = b.m_ranks[sorted_ts];
+
         if (not writetrack[ts])
-          continue;  // Note: this speed up prevents transitive masking (possibly marginal gain).
+          continue;  //FIXME: this speed up prevents transitive masking; check build cost!
```

Comment:
make sure to not revert the existing changes (the same comment about `eta1`->`Eta1` and similar diffs below